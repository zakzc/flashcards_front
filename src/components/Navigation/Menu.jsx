import React, { Component } from "react";
import { NavLink } from "react-router-dom";
// icons
import Pile from "../../assets/icons/pile";
import Play from "../../assets/icons/play";
import EditCard from "../../assets/icons/editCard";
import EditStacks from "../../assets/icons/editStacks";
import LogOut from "../../assets/icons/logOut";

export default class Menu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentUser: props.currentUser,
      userIsLoggedIn: props.userIsLoggedIn,
    };
    // methods from App
    this.logUserOut = this.logUserOut.bind(this);
  }

  logUserOut() {
    this.logOut_User();
  }

  render() {
    return (
      <div id="LogInNavBar">
        <ul
          style={{ listStyleType: "none", padding: 0 }}
          activeclassname="active"
          id="menuBar"
        >
          <li className="navigationLink ">
            <NavLink to="/curStack">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <Pile />
                  </div>
                  <div className="flipMenuBack">Current Stack</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/cardSet">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <Play />
                  </div>
                  <div className="flipMenuBack">Play stack</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/cardManagement">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <EditCard />
                  </div>
                  <div className="flipMenuBack">Manage cards</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/stackManagement">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <EditStacks />
                  </div>{" "}
                  <div className="flipMenuBack">Manage stacks</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <LogOut />
                  </div>
                  <div className="flipMenuBack" onClick={this.logUserOut}>
                    Log Out
                  </div>
                </div>
              </div>
            </NavLink>
          </li>
        </ul>
      </div>
    );
  }
}
