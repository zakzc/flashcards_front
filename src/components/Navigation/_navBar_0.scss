.active {
  .flipMenuFront {
    color: $mainColor;
  }
}

#logOutButton {
  border-radius: 20px;
  margin-left: 2%;
}

.logOutIcon {
  background-color: $contrastColor;
  color: $backColorNav;
  // transform: rotate(180deg);
}

//Links
a {
  color: $backColorNav;
  text-align: center;
  text-decoration: none;
  // display: inline-block;
}

