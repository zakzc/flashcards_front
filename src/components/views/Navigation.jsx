import React, { Component } from "react";
import { Link, NavLink } from "react-router-dom";
// icons
import Pile from "../icons/pile";
import Play from "../icons/play";
import EditCard from "../icons/editCard";
import EditStacks from "../icons/editStacks";
import LogOut from "../icons/logOut";
import LogIn from "../icons/logIn";

export default class Navigation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentUser: props.currentUser,
      userIsLoggedIn: props.userIsLoggedIn,
    };
    // methods from App
    this.logOut_User = this.props.logOut_User;
    this.loggedInNavBar = this.loggedInNavBar.bind(this);
    this.loggedOutNavBar = this.loggedOutNavBar.bind(this);
    // Methods
    // this.logInLogOut = props.logInLogOut();
  }

  logUserOut() {
    console.log("Log me out");
    this.setState({
      currentUser: "",
      userIsLoggedIn: false,
    });
    this.logOut_User();
  }

  logInButton() {
    return (
      <React.Fragment>
        <button
          id="logOutButton"
          type="button"
          onClick={() => {
            this.logUserOut();
          }}
        ></button>
      </React.Fragment>
    );
  }

  loggedInNavBar() {
    return (
      <div id="NavigationMenu">
        <ul
          style={{ listStyleType: "none", padding: 0 }}
          activeclassname="active"
        >
          <li className="navigationLink ">
            <NavLink to="/curStack">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <Pile />
                  </div>
                  <div className="flipMenuBack">Current Stack</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/cardSet">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <Play />
                  </div>
                  <div className="flipMenuBack">Play stack</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/cardManagement">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <EditCard />
                  </div>
                  <div className="flipMenuBack">Manage cards</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/stackManagement">
              <div className="flipMenu">
                <div className="flipMenuInner">
                  <div className="flipMenuFront">
                    <EditStacks />
                  </div>{" "}
                  <div className="flipMenuBack">Manage stacks</div>
                </div>
              </div>
            </NavLink>
          </li>
          <li className="navigationLink">
            <NavLink to="/logIn">
              <div className="flipMenuInner">
                <div className="flipMenu">
                  <div className="flipMenuFront">
                    <LogOut className="logOutIcon" />
                  </div>{" "}
                  <div className="flipMenuBack" onClick={this.logUserOut}>
                    Log Out
                  </div>
                </div>{" "}
              </div>
            </NavLink>
          </li>
        </ul>
      </div>
    );
  }

  loggedOutNavBar() {
    return (
      <React.Fragment>
        <ul
          style={{ listStyleType: "none", padding: 0 }}
          activeclassname="active"
        >
          <li className="navigationLink">
            <Link to="/logIn" className="flipMenuInner">
              <div className="flipMenuFront">
                <LogIn className="logInIcon" />
              </div>{" "}
              <div className="flipMenuBack">Log in</div>
            </Link>
          </li>
        </ul>
      </React.Fragment>
    );
  }

  navigationLayout() {
    return (
      <div>
        {this.state.userIsLoggedIn ? (
          <div id="navigation">{this.loggedInNavBar()}</div>
        ) : (
          <div id="navigation">{this.loggedOutNavBar()}</div>
        )}
      </div>
    );
  }

  render() {
    return <React.Fragment>{this.navigationLayout()}</React.Fragment>;
  }
}
