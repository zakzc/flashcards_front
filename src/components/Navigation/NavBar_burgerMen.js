import React, { useState } from "react";
import Menu from "./Menu";

export default function Burger() {
  const [burgerToggle, setBurgerToggle] = useState(false);
  // const [burgerView, setBurgerView] = useState("closed");

  return (
    <div id="burger">
      <button
        className={burgerToggle ? "change" : "closed"}
        onClick={() => setBurgerToggle(!burgerToggle)}
      >
        <div className="bar1"></div>
        <div className="bar2"></div>
        <div className="bar3"></div>
      </button>
      {burgerToggle && <Menu />}
    </div>
  );
}
// TODO: find a way of displaying the actual menubar.
