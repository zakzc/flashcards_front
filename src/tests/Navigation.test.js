import React from "react";
import Enzyme from "enzyme";
///
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";
///
import Navigation from "../components/Navigation/NavBar_0";

Enzyme.configure({ adapter: new Adapter() });

// Navigation
describe("\n __| Navigation |__", () => {
  const navigation = shallow(<Navigation />);

  describe("Initialization process", () => {
    it("Renders the Navigation", () => {
      expect(navigation).toMatchSnapshot();
    });

    describe("Imports", () => {
      describe("Svg imports", () => {
        test("Pile", () => {
          expect(navigation.find("Pile")).toBeTruthy();
        });
        test("Play", () => {
          expect(navigation.find("Play")).toBeTruthy();
        });
        test("EditCard", () => {
          expect(navigation.find("EditCard")).toBeTruthy();
        });
        test("EditStacks", () => {
          expect(navigation.find("EditStacks")).toBeTruthy();
        });
        test("LogOut", () => {
          expect(navigation.find("LogOut")).toBeTruthy();
        });
        test("LogIn", () => {
          expect(navigation.find("LogIn")).toBeTruthy();
        });
      });
    });
  });
});

// Logged in navigation
// const navLogInProps = {
//   currentUser: {
//     firstName: "Test",
//   },
//   userIsLoggedIn: true,
// };
// const NavBar = shallow(<Navigation {...navLogInProps} />);

// describe("Test different Views", () => {
//   test("Logged In NavBar on true for user logged in", () => {
//     const Navigation = NavBar.find("#navigation");
//     console.log("NAV", Navigation);
//     expect(Navigation).toContain("#LogOutNavBar");
//   });
// });
