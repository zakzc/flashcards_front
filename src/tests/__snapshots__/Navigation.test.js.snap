// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
 __| Navigation |__ Initialization process Renders the Navigation 1`] = `ShallowWrapper {}`;
